python-package/lightgbm/engine.py

```diff
@@ -196,8 +196,10 @@ def train(params, train_set, num_boost_round=100,
             break
     if booster.attr('best_iteration') is not None:
         booster.best_iteration = int(booster.attr('best_iteration')) + 1
+        booster.best_score = float(booster.attr('best_score'))
     else:
         booster.best_iteration = -1
+        booster.best_score = -1
```

Comment:
don't use -1 here, -1 can be 'real' best_score; `float('inf')` or `None` might be better